cmd/osm-controller/certificates_test.go

```diff
@@ -14,6 +14,7 @@ import (
 
 	"github.com/openservicemesh/osm/pkg/certificate/pem"
 	"github.com/openservicemesh/osm/pkg/certificate/providers/tresor"
+	"github.com/openservicemesh/osm/pkg/certproviderutil"
```

Comment:
I don't think `certproviderutil` gives me much context around what this is meant to do.  I suspect this package could become the `misc` to cert issuance related code.

What if we instead put all new functions in `github.com/openservicemesh/osm/pkg/certificate` ?

Reply:
Tried this, but it results in import cycles. I moved this under `pkg/certificate/providers`.

```package github.com/openservicemesh/osm/cmd/osm-controller
    imports github.com/openservicemesh/osm/pkg/catalog
    imports github.com/openservicemesh/osm/pkg/certificate
    imports github.com/openservicemesh/osm/pkg/debugger
    imports github.com/openservicemesh/osm/pkg/catalog: import cycle not allowed
```